helix-term/src/ui/picker.rs

```diff
@@ -364,6 +368,43 @@ impl<T> Picker<T> {
         self.cursor = pos;
     }
 
+    pub fn page_up(&mut self) {
+        if self.matches.is_empty() {
+            return;
+        }
+        let len = self.matches.len();
+        let pos = ((self.cursor + len.saturating_sub(self.completion_height as usize)) % len) % len;
```

Comment:
I think `%` twice here is actually unnecessary, I made a mistake in `move_up`. Feel free to simplify there too

Reply:
Yes, I think I'll use `rem_euclid` here. It's available since Rust 1.38.

Reply:
I don't think it makes a difference here, per docs:

> rem_euclid:
> Since, for the positive integers, all common definitions of division are equal, this is exactly equal to self % rhs.

Reply:
@archseer I generalized the code, it now has a method to scroll by an arbitrary number of lines, positive or negative. In this case, `rem_euclid` is useful.